featurebyte/config.py

```diff
@@ -434,21 +469,22 @@ def use_profile(cls, profile_name: str) -> None:
 
         >>> fb.Configurations().use_profile("local")
         """
-        profile_names = [profile.name for profile in Configurations().profiles or []]
-        if profile_name not in profile_names:
+        new_profile: Optional[Profile] = None
+        for prof in self.profiles:
+            if prof.name == profile_name:
+                new_profile = prof
+                break
+        if new_profile is None:
             raise InvalidSettingsError(f"Profile not found: {profile_name}")
 
         # test connection
-        current_profile_name = os.environ.get("FEATUREBYTE_PROFILE")
+        old_profile = self._profile
         try:
-            os.environ["FEATUREBYTE_PROFILE"] = profile_name
-            cls.check_sdk_versions()
+            self._profile = new_profile
+            self.check_sdk_versions()
         except InvalidSettingsError:
             # restore previous profile
-            if current_profile_name:
-                os.environ["FEATUREBYTE_PROFILE"] = current_profile_name
-            else:
-                os.environ.pop("FEATUREBYTE_PROFILE", None)
+            self._profile = old_profile
```

Comment:
No need to maintain `FEATUREBYTE_PROFILE` env anymore. Every Configurations().profile would have a singluar value.